homeassistant/components/livisi/coordinator.py

```diff
@@ -110,18 +122,29 @@ async def async_set_all_rooms(self) -> None:
 
     def on_data(self, event_data: LivisiEvent) -> None:
         """Define a handler to fire when the data is received."""
-        self._async_dispatcher_send(
-            LIVISI_STATE_CHANGE, event_data.source, event_data.onState
-        )
-        self._async_dispatcher_send(
-            LIVISI_STATE_CHANGE, event_data.source, event_data.vrccData
-        )
-        self._async_dispatcher_send(
-            LIVISI_REACHABILITY_CHANGE, event_data.source, event_data.isReachable
-        )
-        self._async_dispatcher_send(
-            LIVISI_STATE_CHANGE, event_data.source, event_data.isOpen
-        )
+        if event_data.type == LIVISI_EVENT_BUTTON_PRESSED:
+            device_id = self.capability_to_device.get(event_data.source)
+            if device_id is not None:
+                livisi_event_data = {
+                    "device_id": device_id,
+                    "type": EVENT_BUTTON_PRESSED,
+                    "button_index": event_data.properties.get("index", 0),
+                    "press_type": event_data.properties.get("type", "ShortPress"),
+                }
+                self.hass.bus.async_fire("livisi_event", livisi_event_data)
```

Comment:
We've added a new entity, that is a good fit for this: The `event` entity.

See: <https://developers.home-assistant.io/docs/core/entity/event>

I would recommend implementing that instead.